### PR TITLE
IGNITE-19072 Refactoring of org.apache.ignite.internal.index.IndexManagerTest

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-19072

### CODE REVIEW

File: modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java [hunk 1]
```diff
@@ -53,15 +53,18 @@
      * Class of throwable that should be the cause of fail if the future should fail. If {@code null}, the future should be completed
      * successfully.
      */
-    private final Class<? extends Throwable> causeOfFail;
+    private final @Nullable Class<? extends Throwable> causeOfFail;
+
+    /** Fragment that must be a substring of a error message (if {@code null}, message won't be checked). */
```
Comment (ibessonov) [inline modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java:58]:
```suggestion
    /** Fragment that must be a substring of an error message (if {@code null}, message won't be checked). */
```

File: modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java [hunk 2]
```diff
@@ -70,22 +73,23 @@ private CompletableFutureMatcher(Matcher<T> matcher) {
      * @param matcher Matcher to forward the result of the completable future.
      * @param timeout Timeout.
      * @param timeoutTimeUnit {@link TimeUnit} for timeout.
-     * @param causeOfFail If {@code null}, the future should be completed successfully, otherwise it specifies the class of cause
-     *                    throwable.
+     * @param causeOfFail If {@code null}, the future should be completed successfully, otherwise it specifies class of cause throwable.
```
Comment (ibessonov) [inline modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java:76]: Why did you remove "the"?

File: modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java [hunk 3]
```diff
@@ -70,22 +73,23 @@ private CompletableFutureMatcher(Matcher<T> matcher) {
...
+     * @param errorMessageFragment Fragment that must be a substring of a error message (if {@code null}, message won't be checked).
```
Comment (ibessonov) [inline modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java:77]:
```suggestion
     * @param errorMessageFragment Fragment that must be a substring of an error message (if {@code null}, message won't be checked).
```

Comment (ibessonov) [reply modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java:77]: You get the point

File: modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java [hunk 4]
```diff
@@ -53,15 +53,18 @@
      * Class of throwable that should be the cause of fail if the future should fail. If {@code null}, the future should be completed
      * successfully.
      */
-    private final Class<? extends Throwable> causeOfFail;
+    private final @Nullable Class<? extends Throwable> causeOfFail;
+
+    /** Fragment that must be a substring of a error message (if {@code null}, message won't be checked). */
+    private final @Nullable String errorMessageFragment;
```
Comment (ibessonov) [inline modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java:59]: Since you added a new parameter, please add it into `describeMismatchSafely` or other appropriate place

File: modules/index/src/test/java/org/apache/ignite/internal/index/IndexManagerTest.java [hunk 5]
```diff
@@ -92,100 +64,57 @@
...
+            return completedFuture(new TableImpl(tbl, new HeapLockManager(), () -> completedFuture(List.of())));
         });
 
         SchemaManager schManager = mock(SchemaManager.class);
-        when(schManager.schemaRegistry(anyLong(), any())).thenReturn(CompletableFuture.completedFuture(null));
+
+        when(schManager.schemaRegistry(anyLong(), any())).thenReturn(completedFuture(null));
 
         indexManager = new IndexManager(tablesConfig, schManager, tableManagerMock);
         indexManager.start();
```
Comment (ibessonov) [inline modules/index/src/test/java/org/apache/ignite/internal/index/IndexManagerTest.java:95]: Shouldn't you stop the manager in "tearDown"?

File: modules/schema/src/test/java/org/apache/ignite/internal/schema/configuration/index/IndexValidatorImplTest.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * For {@link IndexValidatorImpl} testing.
+ */
+@ExtendWith(ConfigurationExtension.class)
+public class IndexValidatorImplTest {
+    @InjectConfiguration(
+            value = "mock.tables.fooTable {columns.column0 {type.type: STRING}}",
+            internalExtensions = ExtendedTableConfigurationSchema.class
```
Comment (ibessonov) [inline modules/schema/src/test/java/org/apache/ignite/internal/schema/configuration/index/IndexValidatorImplTest.java:46]: Is this necessary? I believe that all extensions are loaded implicitly

**Review by ibessonov — APPROVED** [review]
I think I'm good with these changes, thank you!

File: modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java (hunk 1 above)
Comment (tkalkirill) [reply modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java:58]: Fix it

File: modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java (hunk 2 above)
Comment (tkalkirill) [reply modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java:76]: Fix it

File: modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java (hunk 3 above)
Comment (tkalkirill) [reply modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java:77]: fix it

File: modules/index/src/test/java/org/apache/ignite/internal/index/IndexManagerTest.java (hunk 5 above)
Comment (tkalkirill) [reply modules/index/src/test/java/org/apache/ignite/internal/index/IndexManagerTest.java:95]: Fix it

File: modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java (hunk 4 above)
Comment (tkalkirill) [reply modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/matchers/CompletableFutureMatcher.java:59]: Try to fix it.

File: modules/schema/src/test/java/org/apache/ignite/internal/schema/configuration/index/IndexValidatorImplTest.java (hunk 6 above)
Comment (tkalkirill) [reply modules/schema/src/test/java/org/apache/ignite/internal/schema/configuration/index/IndexValidatorImplTest.java]: Tried to fix it.

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogServiceSelfTest.java [hunk 7]
```diff
@@ -156,6 +157,6 @@ public void testCreateTableIfExistsFlag() {
                         .ifTableExists(false)
                         .build());
 
-        assertThat(fut, CompletableFutureMatcher.willFailFast(TableAlreadyExistsException.class));
+        assertThat(fut, CompletableFutureExceptionMatcher.willThrowFast(TableAlreadyExistsException.class));
```
Comment (ibessonov) [inline modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogServiceSelfTest.java:160]: Please use static import

**Review by ibessonov — APPROVED** [review]
I'm good with your changes, thank you!

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogServiceSelfTest.java (hunk 7 above)
Comment (tkalkirill) [reply modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogServiceSelfTest.java]: Fix it
